### ISSUE TITLE
How to Upload certain files as public and other files as private in same bucket.

### ISSUE THREAD

**Issue by dharmesh07** [metaware/carrierwave-google-storage#25]
@jasdeepsingh , 
If there is no solution for this as stated by you in https://github.com/metaware/carrierwave-google-storage/issues/5 , 

Then can you please provide more details how the initializer would be to achieve this? 

How can we initialize two buckets in CarrierWave?

**Comment by RipTheJacker** [comment]
@dharmesh07 https://github.com/carrierwaveuploader/carrierwave/wiki/How-to:-Define-different-storage-configuration-for-each-Uploader.

For this gem the key is `gcloud_storage`.

**Comment by jasdeepsingh** [comment]
@RipTheJacker The link above is broken due to a dot at the end - I embedded the Wiki link in here: https://bit.ly/2E5uwEp

@dharmesh07 I just did a little bit of research and I believe we can add this support to this gem by utilizing the ACL functionality in the underlying `google-cloud-storage` gem. Relevant code doc here: http://googleapis.github.io/google-cloud-ruby/docs/google-cloud-storage/latest/Google/Cloud/Storage/Bucket.html#create_file-instance_method

I can try and take a stab at implementing this functionality - But I cannot commit to any timelines. If you would like to propose a PR I can see this getting implemented rather quickly.
